### PR TITLE
Implement MS20089: ACs playing avatar recordings now correctly handle playAvatarSound

### PR DESCRIPTION
**Pull request by zfox23**
Fixes [MS20089](https://highfidelity.fogbugz.com/f/cases/20089/make-agent-behave-appropriately-if-asked-to-playAvatarSound-while-already-playing-a-Recording).

The behavior here is now:

1. AC begins playback of a `.hfr` recording
2. AC is asked to play a new sound via `Agent.playAvatarSound()`
3. AC will continue visually playing back the avatar recording, but will play the audio from `playAvatarSound()` _instead of_ the audio from the recording.

I created a version of `playRecordingAC.js` that'll accept a specific message from a client, instructing it to play audio. [Here is that modified script](https://www.dropbox.com/s/ttkdikjx7tqvtc4/playRecordingAC_modified.js?dl=0) that you can use during QA testing.

Here's the script that you can run from the Interface Scripting Console to interrupt the recording:
```
var message = {};
var PLAY_AVATAR_SOUND_TEST_SCRIPT_CHANNEL = "com.highfidelity.playSoundTest";

message.soundURL = "https://hifi-content.s3.amazonaws.com/zfox/temp/song.wav";

Messages.sendMessage(PLAY_AVATAR_SOUND_TEST_SCRIPT_CHANNEL, JSON.stringify(message));
```

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16023/HighFidelity-Beta-PR14604-fce4c25.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16023/HighFidelity-Beta-PR14604-fce4c25.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16023/HighFidelity-Beta-Interface-PR14604-fce4c25.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16023/HighFidelity-Beta-PR14604-fce4c25.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16023/HighFidelity-Beta-Interface-PR14604-fce4c25.exe

**Comment by antoninahifi** [comment]
currently testing

**Comment by antoninahifi** [comment]
Pass

File: assignment-client/src/Agent.h [hunk 1]
```diff
@@ -107,6 +107,7 @@ private slots:
     ResourceRequest* _pendingScriptRequest { nullptr };
     bool _isListeningToAudioStream = false;
     SharedSoundPointer _avatarSound;
+    bool _shouldMuteRecordingAudio{ false };
```
Comment (birarda) [inline assignment-client/src/Agent.h]: should be a space before the `{` here

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16039/HighFidelity-Beta-PR14604-f31d685.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16039/HighFidelity-Beta-PR14604-f31d685.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16039/HighFidelity-Beta-Interface-PR14604-f31d685.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16039/HighFidelity-Beta-PR14604-f31d685.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16039/HighFidelity-Beta-Interface-PR14604-f31d685.exe

**Pull request merged**